nemo_text_processing/inverse_text_normalization/en/data/swear_sequences.tsv

```diff
@@ -0,0 +1,436 @@
+2g1c	2***
```

Comment:
Instead of doing this, you can just have the left side and compose with this:

>>> bowdlerize = NEMO_ALPHA + pynini.closure(pynini.cross(NEMO_SPACE | NEMO_ALPHA, "*"), 1)
>>> ("flubble" @ bowdlerize).string()
'f******'

Reply:
```
bowdlerize = (NEMO_ALPHA | NEMO_DIGIT) + pynini.closure(pynini.cross(NEMO_SPACE | NEMO_ALPHA | NEMO_DIGIT, ""), 1)
```

Sorry, missed that there were digits on that line

Reply:
Yes, this should also work. I'll make the change.

But I would like to clarify on this, wouldn't having the data with the transformations be better than performing some operations on the input text and then getting the written form in terms of computation time? Although the difference may not be significantly large, I'm curious to know.

Reply:
Sure, there's some overhead to it, but at the time it's invoked there are so many other compositions happening that one more isn't likely to make a huge amount of difference.

Reply:
Understood, thanks for clarifying.

Reply:
Understood. Thanks for clarifying.

Reply:
Now, I recognise that there are people who have a legitimate use case for this sort of thing, and for them, a simpler form of input is better, which is why I made the suggestion. (Better still, I think, would be if the input file were configurable, like the whitelist file is.) But don't take that for some sort of approval of this sort of censorship mechanism.

Reply:
I hope you mean that the predefined list is not always applicable. Therefore, a user-configurable input file would be an additional option. It originally seemed a good idea to add all common profane words to the list that would always be censored. You're right, it could also be based on user input depending on the individual's use cases.

> But don't take that for some sort of approval of this sort of censorship mechanism.

I couldn't quite understand this. By this do you mean, using FSTs for filtering profane words isn't correct?

Reply:
I'm making no comment about technical correctness, I find censorship mechanisms ideologically abhorrent. But if you must do it, at least make it configurable by the user.

Reply:
Originally, there was only a user-provided flag `filter_profanity` to check if profanity filtering is required or not and is by default marked `True` which can be over-rid by the user. Now additionally, as you suggested a custom file can also be passed by the user.

Reply:
> I'm making no comment about technical correctness, I find censorship mechanisms ideologically abhorrent.

Right, but based on the use case, one can have it enabled/disabled as we have `filter_profanity` to control if we need it or not.

Originally, there was only a user-provided flag `filter_profanity` to check if profanity filtering is required or not and is by default marked `True` which can be over-rid by the user. Now additionally, as you suggested a custom file can also be passed by the user.